types/src/v0/impls/state.rs

```diff
@@ -205,12 +205,17 @@ pub fn validate_proposal(
         });
     }
 
-    if proposal.fee_info().amount() < expected_chain_config.base_fee * block_size {
-        return Err(ProposalValidationError::InsufficientFee {
-            max_block_size: expected_chain_config.max_block_size,
-            base_fee: expected_chain_config.base_fee,
-            proposed_fee: proposal.fee_info().amount(),
-        });
+    // TODO here we are validating the each fee amount is at least
+    // base_fee * block_size. Seems inappropriate to use current block
+    // to calculate fees for bid on future block.
+    for fee_info in proposal.fee_info().clone() {
+        if fee_info.amount() < expected_chain_config.base_fee * block_size {
```

Comment:
With multiple fees is this still the appropriate way to check if fee is sufficient?

Reply:
As discussed in zulip we we need to check against total sum of all amounts.